buildmaster/NMCPD/filter.py

```diff
@@ -0,0 +1,69 @@
+#Filter for NMCPD
+
+import sys
+import yaml
```

Comment:
The function that retrieves the tables in `YAML` format could look like:
```suggestion
import yaml
import requests

from urllib.request import urlopen

def download_hepdata_table(list_dict_tables: list, table_number: int) -> dict:
    # Loop over the various links of the tables
    for table in list_dict_tables:
        if table_number == int(table["name"].split()[-1]):
            table_dict = requests.get(table["@id"]).json()
            # Loop over the various table formats
            for url_tab in table_dict["distribution"]:
                if "YAML" in url_tab["description"]:
                    reads_yaml = urlopen(url_tab["contentUrl"])
    return yaml.safe_load(reads_yaml)
```
And one can choose to still write the raw data into files if needed to.